src/PrestaShopBundle/Service/TranslationService.php

```diff
@@ -148,65 +172,50 @@ private function requiresThemeTranslationsFactory($theme, $type)
      * @param string $locale
      * @param string $domain
      * @param string|null $theme
-     * @param string|null $search
+     * @param array|null $search
      * @param string|null $module
      *
      * @return array
      */
-    public function listDomainTranslation($locale, $domain, $theme = null, $search = null, $module = null)
-    {
-        if (!empty($theme) && 'classic' !== $theme) {
-            $translationProvider = $this->container->get('prestashop.translation.theme_provider');
-            $translationProvider->setThemeName($theme);
-        } else {
-            $translationProvider = $this->container->get('prestashop.translation.search_provider');
-            if ($module !== null && $translationProvider instanceof UseModuleInterface) {
-                $translationProvider->setModuleName($module);
-            }
-        }
+    public function listDomainTranslation(
+        string $locale,
+        string $domain,
+        ?string $theme = null,
+        ?array $search = null,
+        ?string $module = null
+    ): array {
         if ('Messages' === $domain) {
             $domain = 'messages';
```

Comment:
```suggestion
        if ('Messages' === $domain) {
            $domain = 'messages';
```
Both strings could be in const